stdlib/public/Concurrency/MonotonicClock.swift

```diff
@@ -0,0 +1,129 @@
+    
+    init(nanoseconds: UInt64) {
+      self.nanoseconds = nanoseconds
+    }
+    
+    public init?(converting deadline: _Date) {
+      var monotonicNanoseconds = UInt64(0)
+      var realtimeSeconds = Int64(0)
+      var realtimeNanoseconds = Int64(0)
+      _getTime(
+        absoluteNanoseconds: nil,
+        continuousNanoseconds: &monotonicNanoseconds,
+        realtimeSeconds: &realtimeSeconds, realtimeNanoseconds: &realtimeNanoseconds
+      )
+      let monotonicNow = Instant(nanoseconds: monotonicNanoseconds)
+      let realtimeNow = _Date(durationSince1970: .seconds(realtimeSeconds) + .nanoseconds(realtimeNanoseconds))
+      
+      let duration = realtimeNow.duration(to: deadline)
+      if duration > Duration.nanoseconds(Int64.max) {
+        return nil
+      }
+      self = monotonicNow.advanced(by: duration)
+    }
+  }
+}
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, macCatalyst 9999, *)
+extension _Date {
+  public init(converting deadline: MonotonicClock.Instant) {
+    var monotonicNanoseconds = UInt64(0)
+    var realtimeSeconds = Int64(0)
+    var realtimeNanoseconds = Int64(0)
+    _getTime(
+      absoluteNanoseconds: nil,
+      continuousNanoseconds: &monotonicNanoseconds,
+      realtimeSeconds: &realtimeSeconds, realtimeNanoseconds: &realtimeNanoseconds
+    )
+    
+    let realtimeNow = _Date(durationSince1970: .seconds(realtimeSeconds) + .nanoseconds(realtimeNanoseconds))
+    let monotonicNow = MonotonicClock.Instant(nanoseconds: monotonicNanoseconds)
+    let duration = monotonicNow.duration(to: deadline)
+    self = realtimeNow.advanced(by: .nanoseconds(duration.nanoseconds))
+  }
+}
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, macCatalyst 9999, *)
+extension ClockProtocol where Self == MonotonicClock {
+  public static var monotonic: MonotonicClock { return MonotonicClock() }
+}
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, macCatalyst 9999, *)
+extension MonotonicClock: ClockProtocol {
+  public var now: MonotonicClock.Instant {
+    MonotonicClock.now
+  }
+  
+  public static var now: MonotonicClock.Instant {
+    var nanoseconds = UInt64(0)
+    _getTime(
+      absoluteNanoseconds: nil,
+      continuousNanoseconds: &nanoseconds,
+      realtimeSeconds: nil, realtimeNanoseconds: nil
+    )
+    return MonotonicClock.Instant(nanoseconds: nanoseconds)
+  }
+  
+  public func sleep(until deadline: Instant) async throws {
+    try await Task.sleep(until: 0, deadline.nanoseconds, clockId: .monotonic)
+  }
+}
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, macCatalyst 9999, *)
+extension MonotonicClock.Instant: InstantProtocol {
+  public static var now: MonotonicClock.Instant { MonotonicClock.now }
+  
+  public func advanced(by duration: Duration) -> MonotonicClock.Instant {
+    if duration > .zero {
+      return MonotonicClock.Instant(nanoseconds: nanoseconds + duration.seconds.magnitude * 1000000000 + duration.nanoseconds.magnitude)
```

Comment:
Style nit (troughout this PR): The Swift Standard library has a [hard line length limit of 80 characters](https://github.com/apple/swift/blob/main/docs/StandardLibraryProgrammersManual.md#line-breaking).